### PR TITLE
The AWS SQS negative acknowledger usage bug fix

### PR DESCRIPTION
**Pull request by sbodvanski**
Fixes the issue described in https://github.com/micronaut-projects/micronaut-jms/issues/170 .

### CODE REVIEW

File: jms-core/src/main/java/io/micronaut/jms/annotations/Queue.java [hunk 1]
```diff
@@ -82,8 +82,11 @@
      * option cannot be used in conjunction with {@link Queue#executor()}; if
      * both are specified the {@link Queue#executor()} value will be used.
      *
+     * @deprecated since 3.0.0
```
Comment (graemerocher) [inline jms-core/src/main/java/io/micronaut/jms/annotations/Queue.java:85]: describe why it was deprecated in this one and all other deprecated members

File: jms-core/src/main/java/io/micronaut/jms/configuration/JMSTopicListenerMethodProcessor.java [hunk 2]
```diff
@@ -46,9 +46,13 @@ public JMSTopicListenerMethodProcessor(BeanContext beanContext,
...
     protected ExecutorService getExecutorService(AnnotationValue<Topic> value) {
-
         final Optional<String> executorName = value.stringValue("executor");
-        if (executorName.isPresent() && !executorName.get().isEmpty()) {
+
+        if (!executorName.isPresent()) {
+            return null;
+        }
+
+        if (!executorName.get().isEmpty()) {
```
Comment (graemerocher) [inline jms-core/src/main/java/io/micronaut/jms/configuration/JMSTopicListenerMethodProcessor.java:55]: why do we still need this code if the members were deprecated?

File: jms-core/src/main/java/io/micronaut/jms/configuration/JMSQueueListenerMethodProcessor.java [hunk 3]
```diff
@@ -58,21 +58,24 @@ protected ExecutorService getExecutorService(AnnotationValue<Queue> value) {
...
 
+        if (!executorName.isPresent() && !concurrency.isPresent()) {
+            return null;
+        }
+
         if (executorName.isPresent() && !executorName.get().isEmpty()) {
             return beanContext.findBean(ExecutorService.class, Qualifiers.byName(executorName.get()))
                 .orElseThrow(() -> new IllegalStateException(
                     "No ExecutorService bean found with name " + executorName.get()));
         }
```
Comment (graemerocher) [inline jms-core/src/main/java/io/micronaut/jms/configuration/JMSQueueListenerMethodProcessor.java:69]: why do we still need this code if the members were deprecated?

**Review by graemerocher — CHANGES_REQUESTED** [review]

File: jms-core/src/main/java/io/micronaut/jms/configuration/JMSTopicListenerMethodProcessor.java (hunk 2 above)
Comment (sbodvanski) [reply jms-core/src/main/java/io/micronaut/jms/configuration/JMSTopicListenerMethodProcessor.java:55]: it is to allow backward compatibility for the usage of the annotation concurency options before we completely remove concurrency support. If the annotation concurency options are configured, the implementation is utilized as before. If options are not present, the will not be conurency utilization at jms meesagee proccessing level. 

Do you want to say that we should remove the concurency utilization regardles of presence of the deprecated anotation concurency options?

File: jms-core/src/main/java/io/micronaut/jms/configuration/JMSQueueListenerMethodProcessor.java (hunk 3 above)
Comment (sbodvanski) [reply jms-core/src/main/java/io/micronaut/jms/configuration/JMSQueueListenerMethodProcessor.java:69]: it is to allow backward compatibility for the usage of the annotation concurency options before we completely remove concurrency support. If the annotation concurency options are configured, the implementation is utilized as before. If options are not present, the will not be conurency utilization at jms meesagee proccessing level. 

Do you want to say that we should remove the concurency utilization regardles of presence of the deprecated anotation concurency options?

File: jms-core/src/main/java/io/micronaut/jms/configuration/JMSTopicListenerMethodProcessor.java (hunk 2 above)
Comment (graemerocher) [reply jms-core/src/main/java/io/micronaut/jms/configuration/JMSTopicListenerMethodProcessor.java:55]: If we retain the behaviour we should as a minimum log a deprecation warning when it is used

**Comment by sbodvanski** [comment]
Getting the rate limit exceeded error in git tasks :

`Failed to read from Github API. Response code: 403

Response message: Forbidden
> Task :jms-activemq-artemis:findBaseline FAILED
Error body: {"message":"API rate limit exceeded for user ID [65](https://github.com/micronaut-projects/micronaut-jms/actions/runs/4243614061/jobs/7376591646#step:7:66)172877.","documentation_url":"https://docs.github.com/rest/overview/resources-in-the-rest-api#rate-limiting"}
Response headers: {null=[HTTP/1.1 403 Forbidden], X-RateLimit-Resource=[core], Server=[GitHub.com], Access-Control-Allow-Origin=[*], X-RateLimit-Used=[5006], X-Content-Type-Options=[nosniff], X-RateLimit-Reset=[1[67](https://github.com/micronaut-projects/micronaut-jms/actions/runs/4243614061/jobs/7376591646#step:7:68)7077881], Date=[Wed, 22 Feb 2023 14:18:11 GMT], Referrer-Policy=[origin-when-cross-origin, strict-origin-when-cross-origin], X-Frame-Options=[deny], Strict-Transport-Security=[max-age=31536000; includeSubdomains; preload], Access-Control-Expose-Headers=[ETag, Link, Location, Retry-After, X-GitHub-OTP, X-RateLimit-Limit, X-RateLimit-Remaining, X-RateLimit-Used, X-RateLimit-Resource, X-RateLimit-Reset, X-OAuth-Scopes, X-Accepted-OAuth-Scopes, X-Poll-Interval, X-GitHub-Media-Type, X-GitHub-SSO, X-GitHub-Request-Id, Deprecation, Sunset], X-RateLimit-Remaining=[0], X-GitHub-Media-Type=[github.v3; format=json], Content-Security-Policy=[default-src 'none'], Vary=[Accept-Encoding, Accept, X-Requested-With], X-RateLimit-Limit=[5000], X-XSS-Protection=[0], Content-Length=[159], github-authentication-token-expiration=[2024-01-16 00:00:00 +0100], X-GitHub-Request-Id=[7400:7563:2DD151C:2EBBE6F:63F62423], Content-Type=[application/json; charset=utf-8]}`

**Review by graemerocher — APPROVED** [review]

**Pull request merged**